lib/runtime/allocator/freeing_bump.go

```diff
@@ -505,14 +505,17 @@ func (f *FreeingBumpHeapAllocator) Deallocate(mem runtime.Memory, ptr uint32) (e
 func bump(bumper *uint32, size uint32, mem runtime.Memory) (uint32, error) {
 	requiredSize := uint64(*bumper) + uint64(size)
 
-	if requiredSize > uint64(mem.Size()) {
+	if requiredSize > mem.Size() {
 		requiredPages, ok := pagesFromSize(requiredSize)
 		if !ok {
-			return 0, fmt.Errorf("%w: required size %d dont fit uint32",
-				ErrAllocatorOutOfSpace, requiredSize)
+			panic(fmt.Sprintf("cannot calculate number of pages from size %d", requiredSize))
+		}
+
+		currentPages, ok := pagesFromSize(mem.Size())
+		if !ok {
+			panic(fmt.Sprintf("cannot calculate current number of pages, current size: %d", mem.Size()))
```

Comment:
pagesFromSize returns false if number of pages can't fit in to uint32, so I think panic message should correspond to that